datafusion/datasource-parquet/src/file_format.rs

```diff
@@ -1515,13 +1498,11 @@ fn spawn_parquet_parallel_serialization_task(
     SpawnedTask::spawn(async move {
         let max_buffer_rb = parallel_options.max_buffered_record_batches_per_stream;
         let max_row_group_rows = writer_props.max_row_group_size();
+        let mut row_group_index = 0;
+        let col_writers =
```

Comment:
It is really nice to use the `ArrowRowGroupWriterFactory` API here for all writing on both paths 👍